### PR TITLE
Add suport for setting defaultAttribute #LMR-790

### PR DESCRIPTION
**Pull request by kubedo8**

### CODE REVIEW

File: api/src/main/java/io/lumeer/api/dto/JsonCollection.java [hunk 1]
```diff
@@ -102,8 +104,8 @@ public void updateAttribute(final String attributeId, final Attribute attribute)
    }
 
    @Override
-   public void deleteAttribute(final String attributeName) {
-      attributes.removeIf(attribute -> AttributeUtil.isEqualOrChild(attribute, attributeName));
+   public void deleteAttribute(final String attributeId) {
+      attributes.removeIf(attribute -> attribute.getId().equals(attributeId));
```
Comment (livthomas) [inline api/src/main/java/io/lumeer/api/dto/JsonCollection.java]: This change breaks deleting child attributes.

Comment (kubedo8) [reply api/src/main/java/io/lumeer/api/dto/JsonCollection.java]: Really? How do we know which are child attributes?

Comment (livthomas) [reply api/src/main/java/io/lumeer/api/dto/JsonCollection.java]: Those which names start with the name of the deleted attribute + `.`. It was easier before. Now we first need to get the deleted attribute and then remove its children.

Comment (kubedo8) [reply api/src/main/java/io/lumeer/api/dto/JsonCollection.java]: Oh sorry, I misunderstood how we will use `name` field. I will definetely revert this change.

File: api/src/main/java/io/lumeer/api/model/Collection.java [hunk 2]
```diff
@@ -46,7 +46,7 @@ default ResourceType getType() {
 
    void updateAttribute(String attributeId, Attribute attribute);
 
-   void deleteAttribute(String attributeId);
+   void deleteAttribute(String attributeName);
```
Comment (livthomas) [inline api/src/main/java/io/lumeer/api/model/Collection.java]: Don't we pass `attributeId` to this method in `CollectionFacade`. The method interface can stay as you designed it, you just need to change the implementation. Find the attribute to be removed first in order to get the name and then remove it together with its children.

**Pull request merged**